From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jesse Ezell <jezell@users.noreply.github.com>
Date: Sun, 27 Mar 2025 14:49:01 -0800
Subject: Add engine build ci infra

diff --git a/engine/scripts/slim-linux-android.gclient b/engine/scripts/slim-linux-android.gclient
new file mode 100644
index 0000000000..ef6c08d028
--- /dev/null
+++ b/engine/scripts/slim-linux-android.gclient
@@ -0,0 +1,23 @@
# Copy this file to the root of your flutter checkout to bootstrap gclient
# or just run gclient sync in an empty directory with this file.
solutions = [
  {
    "custom_deps": {},
    "deps_file": "DEPS",
    "managed": False,
    "name": ".",
    "safesync_url": "",

    # If you are using SSH to connect to GitHub, change the URL to:
    # git@github.com:flutter/flutter.git
    "url": "https://github.com/flutter/flutter.git",


    "custom_vars": {
        "download_linux_deps": False,
        "download_android_deps": True,
        "download_esbuild": False,
        "download_fuchsia_deps": False,
    }
  },
]
\ No newline at end of file
diff --git a/engine/scripts/slim-linux-desktop.gclient b/engine/scripts/slim-linux-desktop.gclient
new file mode 100644
index 0000000000..b7d27e26fb
--- /dev/null
+++ b/engine/scripts/slim-linux-desktop.gclient
@@ -0,0 +1,23 @@
# Copy this file to the root of your flutter checkout to bootstrap gclient
# or just run gclient sync in an empty directory with this file.
solutions = [
  {
    "custom_deps": {},
    "deps_file": "DEPS",
    "managed": False,
    "name": ".",
    "safesync_url": "",

    # If you are using SSH to connect to GitHub, change the URL to:
    # git@github.com:flutter/flutter.git
    "url": "https://github.com/flutter/flutter.git",


    "custom_vars": {
        "download_linux_deps": True,
        "download_android_deps": False,
        "download_esbuild": False,
        "download_fuchsia_deps": False,
    }
  },
]
\ No newline at end of file
diff --git a/engine/scripts/slim-linux-web.gclient b/engine/scripts/slim-linux-web.gclient
new file mode 100644
index 0000000000..84524ab817
--- /dev/null
+++ b/engine/scripts/slim-linux-web.gclient
 @@ -0,0 +1,24 @@
# Copy this file to the root of your flutter checkout to bootstrap gclient
# or just run gclient sync in an empty directory with this file.
solutions = [
  {
    "custom_deps": {},
    "deps_file": "DEPS",
    "managed": False,
    "name": ".",
    "safesync_url": "",

    # If you are using SSH to connect to GitHub, change the URL to:
    # git@github.com:flutter/flutter.git
    "url": "https://github.com/flutter/flutter.git",


    "custom_vars": {
        "download_linux_deps": True,
        "download_android_deps": False,
        "download_esbuild": True,
        "download_fuchsia_deps": False,
        "download_emsdk": True,
    }
  },
]
\ No newline at end of file
diff --git a/engine/scripts/slim-mac.gclient b/engine/scripts/slim-mac.gclient
new file mode 100644
index 0000000000..38be3fde02
--- /dev/null
+++ b/engine/scripts/slim-mac.gclient
 @@ -0,0 +1,23 @@
# Copy this file to the root of your flutter checkout to bootstrap gclient
# or just run gclient sync in an empty directory with this file.
solutions = [
  {
    "custom_deps": {},
    "deps_file": "DEPS",
    "managed": False,
    "name": ".",
    "safesync_url": "",

    # If you are using SSH to connect to GitHub, change the URL to:
    # git@github.com:flutter/flutter.git
    "url": "https://github.com/flutter/flutter.git",


    "custom_vars": {
        "download_linux_deps": False,
        "download_android_deps": False,
        "download_esbuild": False,
        "download_fuchsia_deps": False,
    }
  },
]
\ No newline at end of file
diff --git a/engine/scripts/slim-windows.gclient b/engine/scripts/slim-windows.gclient
new file mode 100644
index 0000000000..581802c425
--- /dev/null
+++ b/engine/scripts/slim-windows.gclient
 @@ -0,0 +1,22 @@
# Copy this file to the root of your flutter checkout to bootstrap gclient
# or just run gclient sync in an empty directory with this file.
solutions = [
  {
    "custom_deps": {},
    "deps_file": "DEPS",
    "managed": False,
    "name": ".",
    "safesync_url": "",

    # If you are using SSH to connect to GitHub, change the URL to:
    # git@github.com:flutter/flutter.git
    "url": "https://github.com/flutter/flutter.git",


    "custom_vars": {
        "download_android_deps": False,
        "download_esbuild": False,
        "download_fuchsia_deps": False,
    }
  },
]
\ No newline at end of file
